Day_3/day_3.py

```diff
@@ -0,0 +1,428 @@
+import os
+import numpy as np
+
+import math
+
+class Box:
+    def __init__(self, box_length, particles=None):
+        self.box_length = box_length
+        self.particles = particles # if None, no particles in MCSystem
+        
+        # Calculated quantities
+        self.calculate_number_particles()
+    
+    def calculate_number_particles(self):
+        """Calculate the number of particles"""
+        if self.particles is None:
+            self.number_particles = 0
+        else:
+            self.number_particles = len(self.particles)
+    
+    def minimum_image_distance(self, r_i, r_j):
+        """
+        Calculate the distance between two points using the minimum image convention.
+
+        The minimum image convention is for calculating distances in boxes with periodic boundary conditions.
+
+        Parameters
+        ----------
+        r_i : numpy array
+            The coordinates of first point np.array(x,y,z) 
+        r_j : numpy array
+            The coordinates of second point np.array(x,y,z)
+        
+        Returns
+        -------
+        rij2 : float
+            The square distance between two particles.
+        """
+
+        rij = r_i - r_j
+        rij = rij - self.box_length * np.round(rij / self.box_length)
+        rij2 = np.dot(rij, rij)
+        return rij2
+    
+    @property
+    def volume(self):
+        """Calculate the volume of the box based on the box length"""
+        return self.box_length ** 3
+    
+    def wrap(self, r_i):
+        """Wrap points which are outside of box into box"""
+        # TODO
+        pass
+    
+class TriclinicBox(Box):
+    def __init__(self, box_length, tilt_factors):
+        """
+        Create a triclinic box
+        """
+        super().__init__(box_length)
+        self.tilt_factors = tilt_factors
+        self.calculate_box_angles()
+    
+    def calculate_box_angles(self):
+        # TODO - implement this
+        self.box_angles = self.tilt_factors
+        pass
+
+    @property
+    def volume(self):
+        term1 = self.box_length ** 3
+        term1 *= 1 - math.cos(self.box_angles[0]) ** 2 - math.cos(self.box_angles[1]) ** 2 - math.cos(self.box_angles[2]) ** 2
+        term2 = 2 * math.sqrt(math.cos(self.box_angles[0])*math.cos(self.box_angles[1])*math.cos(self.box_angles[2]))
+        return term1 + term2
+
+
+class MCState:
+    def __init__(self, box, cutoff, max_displacement, reduced_temperature):
+        self.box = box
+        self.cutoff = cutoff
+        self.beta = 1 / reduced_temperature
+        self.max_displacement = max_displacement
+        self.coordinates = box.particles
```

Comment:
This was done out of laziness to avoid changing more of the code. I can change this in the implementation, but didn't view it as very important because we will not be providing this solution to students.